src/clp_ffi_js/ir/LogEventWithFilterData.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+#define CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+
+#include <concepts>
+#include <utility>
+
+#include <clp/ir/LogEvent.hpp>
+#include <clp/ir/types.hpp>
+
+#include <clp_ffi_js/constants.hpp>
+
+namespace clp_ffi_js::ir {
+using clp::ir::four_byte_encoded_variable_t;
+using UnstructuredLogEvent = clp::ir::LogEvent<four_byte_encoded_variable_t>;
+
+// Concept defining valid log event types.
+// TODO: Extend valid log event types when filtering support is added for structured logs.
+template <typename T>
+concept ValidLogEventTypes = std::same_as<T, UnstructuredLogEvent>;
```

Comment:
Why not inline the concept into the requires statement? We don't have formal guidelines on using concepts yet, but I feel like we can avoid creating a concept for cases like this.